### PR TITLE
[hailtop] Remove namespace options in hailctl auth commands

### PR DESCRIPTION
**Pull request by daniel-goldstein**
I'm trying to move our codebase away from relying on the notion of a K8s namespace for routing because this model does not hold in alternative deployment environments such as Terra. Further, having a `-n` option in `hailctl auth` commands is awkward because it can only be used for dev environments yet is visible to users in the help menu. As such, this is a breaking change only for developers. The functionality is not really gone though because you can replace any `hailctl auth … -n dgoldste` with `HAIL_DEFAULT_NAMESPACE=dgoldste hailctl auth …`.

### CODE REVIEW

File: hail/python/hailtop/auth/auth.py [hunk 1]
```diff
@@ -266,26 +248,16 @@ async def async_create_user(
...
-    async with httpx.client_session(
-            timeout=aiohttp.ClientTimeout(total=30),
-            headers=headers) as session:
-        await retry_transient_errors(
-            session.post,
-            deploy_config.url('auth', f'/api/v1alpha/users/{username}/create'),
-            json=body
-        )
+    url = get_deploy_config().url('auth', f'/api/v1alpha/users/{username}/create')
+    async with hail_session() as session:
```
Comment (danking) [inline hail/python/hailtop/auth/auth.py:252]: I think we maybe purposely gave extra timeout for this API call.

File: hail/python/hailtop/auth/auth.py [hunk 2]
```diff
@@ -266,26 +248,16 @@ async def async_create_user(
...
-    deploy_config, headers, _ = await deploy_config_and_headers_from_namespace(namespace)
-    async with httpx.client_session(
-            timeout=aiohttp.ClientTimeout(total=300),
-            headers=headers) as session:
-        await retry_transient_errors(
-            session.delete,
-            deploy_config.url('auth', f'/api/v1alpha/users/{username}')
-        )
+async def async_delete_user(username: str):
+    url = get_deploy_config().url('auth', f'/api/v1alpha/users/{username}')
```
Comment (danking) [inline hail/python/hailtop/auth/auth.py:261]: is this API call fast?

**Review by danking — DISMISSED** [review]

File: hail/python/hailtop/auth/auth.py (hunk 1 above)
Comment (daniel-goldstein) [reply hail/python/hailtop/auth/auth.py]: I forgot these weren't immediate operations. I restored the timeouts.

File: hail/python/hailtop/auth/auth.py (hunk 2 above)
Comment (daniel-goldstein) [reply hail/python/hailtop/auth/auth.py:261]: restored

**Review by danking — APPROVED** [review]

**Pull request merged**